src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -53,7 +53,12 @@ case class CsvRelation protected[spark] (
   extends BaseRelation with TableScan with PrunedScan with InsertableRelation {
 
   // Share date format object as it is expensive to parse date pattern.
-  private val dateFormatter = if (dateFormat != null) new SimpleDateFormat(dateFormat) else null
+  private val dateFormatter: Seq[SimpleDateFormat] =
+    if (dateFormat != null) {
+      dateFormat.split(",").map(pattern => new SimpleDateFormat(pattern))
```

Comment:
I can think of a couple of choices here.  Let me know if you have a preference.   Perhaps option b helps to preserve binary compatibility so would be favored?

a - Have another option, dateFormatDelimiter, defaults to ",".  
b - Have delimiter as first part of dateFormat.  Something like "//[delimiter]\n", ala String Calculator kata, http://osherove.com/tdd-kata-1/.  If there is no //[delimiter]\n", the delimiter defaults to ","